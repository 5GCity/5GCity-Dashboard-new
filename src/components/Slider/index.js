/**
 * Slider Component
 * Please write a description
 *
 * @author Your Name <youremail@ubiwhere.com>
 */

import styled from 'styled-components'
import { Slider } from 'element-react'
import { lighten } from 'polished'

export default styled(Slider)`

.el-slider__button-wrapper{
    height:16px;
    width:16px;
    top: -12px;
}

.el-slider__runway{
    background-color: #89979F;
    height: 2px;
}

.el-slider__bar {
    background-color: transparent;
    height:2px;
}

.el-slider__button {
    height: 16px;
    width:16px;
    border: 2px solid #37474F;
    background-color: rgba(140,193,78,0.6);
}

.el-slider__stop {
    height: 6px;
    width: 6px;
    top:-2px;
    background-color: #89979F;
}

.el-tooltip__popper.is-dark {
    background: ${lighten(0.1, '#89979F')};
    color:#fff;
}

.el-tooltip__popper[x-placement^=bottom] .popper__arrow {
    border-bottom-color: ${lighten(0.1, '#89979F')};

}
.el-tooltip__popper[x-placement^=bottom] .popper__arrow::after {
    border-bottom-color: ${lighten(0.1, '#89979F')};
}
`
