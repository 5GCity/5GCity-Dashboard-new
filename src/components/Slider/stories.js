/**
 * Slider Component Stories
 * Please write a description or remove this line
 *
 * @author Your Name <youremail@ubiwhere.com>
 */
import React from 'react'
import { storiesOf } from '@storybook/react'
import { Theme } from 'globalStyles'

import Slider from './index'

const bStyle = {
  padding: '30px',
  backgroundColor: Theme.bodyBackground,
  height: '100%'
}

const example = () => (
  <div style={bStyle}>
    <Slider />
  </div>
)

const exampleMultiplePoints = () => (
  <div style={bStyle}>
    <Slider step='10' showStops />
  </div>
)

storiesOf('Slider', module)
  .add('default', example)
  .add('default + Mutiple points', exampleMultiplePoints)
