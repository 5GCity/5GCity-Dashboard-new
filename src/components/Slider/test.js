/**
 * Slider Component Tests
 *
 * @author Your Name <youremail@ubiwhere.com>
 */
import React from 'react'
import renderer from 'react-test-renderer'
import Slider from './index'
import 'jest-styled-components'

it('renders correctly', () => {
  const tree = renderer
    .create(<Slider />)
    .toJSON()

  expect(tree).toMatchSnapshot()
})
