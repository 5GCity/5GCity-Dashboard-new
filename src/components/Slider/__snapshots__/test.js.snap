// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 .el-slider__button-wrapper {
  height: 16px;
  width: 16px;
  top: -12px;
}

.c0 .el-slider__runway {
  background-color: #89979F;
  height: 2px;
}

.c0 .el-slider__bar {
  background-color: transparent;
  height: 2px;
}

.c0 .el-slider__button {
  height: 16px;
  width: 16px;
  border: 2px solid #37474F;
  background-color: rgba(140,193,78,0.6);
}

.c0 .el-slider__stop {
  height: 6px;
  width: 6px;
  top: -2px;
  background-color: #89979F;
}

.c0 .el-tooltip__popper.is-dark {
  background: #a5b0b6;
  color: #fff;
}

.c0 .el-tooltip__popper[x-placement^=bottom] .popper__arrow {
  border-bottom-color: #a5b0b6;
}

.c0 .el-tooltip__popper[x-placement^=bottom] .popper__arrow::after {
  border-bottom-color: #a5b0b6;
}

<div
  className="el-slider c0"
>
  <div
    className="el-slider__runway"
    onClick={[Function]}
    style={Object {}}
  >
    <div
      className="el-slider__bar"
      style={
        Object {
          "left": "0%",
          "width": "0%",
        }
      }
    />
    <div
      className="el-slider__button-wrapper"
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "left": "0%",
        }
      }
    >
      <div
        className="el-tooltip"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={Object {}}
      >
        <div
          className="el-tooltip__rel"
        >
          <div>
            <div
              className="el-slider__button"
            />
          </div>
        </div>
        <div
          className="el-tooltip__popper is-dark"
          style={
            Object {
              "display": "none",
            }
          }
        >
          <div>
            <span>
              0
            </span>
          </div>
          <div
            className="popper__arrow"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
